app/scripts/modules/core/src/entityTag/EntitySource.tsx

```diff
@@ -1,6 +1,76 @@
-import { IEntityTag } from 'core/domain';
+import { UISref } from '@uirouter/react';
+import { UIRouterContextComponent } from '@uirouter/react-hybrid';
+import * as React from 'react';
+
+import { ICreationMetadataTag, IExecution } from 'core/domain';
+import { HoverablePopover, useData } from 'core/presentation';
+import { ReactInjector } from 'core/reactShims';
+
+import { EntitySourcePopover } from './EntitySourcePopover';
 
 export interface IEntitySourceProps {
-  metadata: IEntityTag;
+  metadata: ICreationMetadataTag;
   relativePath?: string;
 }
+
+export const EntitySource = ({ metadata, relativePath = '^.^.^' }: IEntitySourceProps) => {
+  const executionType = metadata?.value?.executionType === 'pipeline' ? 'pipeline' : 'task';
+  const { application, comments, executionId, stageId } = metadata?.value || {};
+
+  const fetchExecution = () => {
+    if (executionType === 'pipeline') {
+      return ReactInjector.executionService.getExecution(metadata?.value?.executionId);
+    }
+
+    return new Promise(() => {});
+  };
+
+  const { result: execution, status, error } = useData(fetchExecution, undefined, [
+    executionType,
+    metadata?.value?.executionId,
+  ]);
+  const isLoading = status !== 'RESOLVED';
+
+  const srefPath =
+    executionType === 'pipeline'
+      ? `${relativePath}.pipelines.executionDetails.execution`
+      : `${relativePath}.tasks.taskDetails`;
+  const srefParams =
+    executionType === 'pipeline' ? { application, executionId, stageId } : { application, taskId: executionId };
+
+  const PopoverContent = React.useCallback(
+    () => (
+      <EntitySourcePopover
+        comments={comments}
+        execution={execution as IExecution}
+        executionType={executionType}
+        metadata={metadata.value}
+      />
+    ),
+    [executionId],
+  );
```

Comment:
Curious on the use of `useCallback` here. I'm wondering if we just define `PopoverContent` as a react element, React may do the right thing.

Also, is `executionId` a good proxy for determining a change in the props supplied to `EntitySourcePopover`?